app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list[dict]) -> list[Person]:
+
+    for person_dict in people_list:
+        Person(person_dict["name"], person_dict["age"])
+
+    for person_dict in people_list:
+        person_instance = Person.people[person_dict["name"]]
```

Comment:
This line retrieves a `Person` instance from the `people` dictionary. Ensure that the name exists in the dictionary to avoid KeyError.